api/transport/errors.go

```diff
@@ -37,6 +37,13 @@ func IsBadRequestError(err error) bool {
 	return ok
 }
 
+// IsNotFoundError returns true if the request could not be processed
+// because the receiving service did not handle the procedure.
+func IsNotFoundError(err error) bool {
```

Comment:
Consider calling this IsInboundNotFoundError to keep with the
InboundBadRequestError convention. This will only work on the server-side on
`Choose()` calls. The way this is set up right now, one might believe that
this will work:

```go
_, err := client.GetValue(...)
if transport.IsNotFoundError(err) {
  // ...
}
```

Perhaps we should expose UnrecognizedProcedureError in the transport package.
The contract for [`transport.Router`][1] states that it returns an
UnrecognizedProcedureError if the procedure is not found but there's no public
way to build such an error.

If we return a matcher, it would be ideal if there was a way for users to
construct an error that satisfies the matcher, at least for testing. We could
expose a dummy version under the transporttest package if we're reluctant to
commit to the UnrecognizedProcedureError constructor's API.

[1]: https://godoc.org/go.uber.org/yarpc/api/transport#Router